src/vm/runners/cairo_runner.rs

```diff
@@ -297,6 +297,22 @@ impl CairoRunner {
         Ok(())
     }
 
+    /// Mark a memory address as accesed.
+    pub fn mark_as_accessed(
+        &mut self,
+        address: Relocatable,
+        size: usize,
+        vm: &mut VirtualMachine,
+    ) -> Result<(), VirtualMachineError> {
+        let accessed_addressess = match &mut vm.accessed_addresses {
+            Some(x) => x,
+            None => return Err(VirtualMachineError::RunIsNotEnded),
```

Comment:
This error doesn't make sense, it would be better to return something similar to "mark_as_accessed called with no accessed_addresses"

Reply:
Looking at the source code there is misunderstanding, `cairo_runner.accessed_addresses` contains the relocated addresses, and not the vm's `accessed_addresses`. So the error returned is correct, but we should add the field accessed_addresses (or even better, relocated_accessed_addresses to avoid future confusion) to CairoRunner (this field should have the type `Option<Vec<BigInt>>`, as it contains relocated addresses) and use `self.accessed_addresses `inside the method instead of `self.vm.accessed_adresses`

Reply:
Nit: `RunNotFinished`.

Reply:
Alright, it should be fixed now.